scipy/ndimage/src/ni_support.c

```diff
@@ -347,10 +347,14 @@ int NI_ArrayToLineBuffer(NI_LineBuffer *buffer,
                                    pa, pb, length, buffer->line_stride);
             CASE_COPY_DATA_TO_LINE(NPY_LONGLONG, npy_longlong,
                                    pa, pb, length, buffer->line_stride);
+            CASE_COPY_DATA_TO_LINE(NPY_HALF, npy_half,
+                                   pa, pb, length, buffer->line_stride);
```

Comment:
This is wrong: note that the macro expands to `(double)*(_type *)_pi` i.e. `(double)*(npy_half *) <-> (double)*(npy_uint16 *)` which is not the correct way to cast npy_half to a float --- it would be necessary to use the `npy_half_to_double` and `npy_double_to_half` half functions.